core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -664,23 +664,28 @@ public boolean isCoordinator() {
          assertIsNotTerminated();
          return cacheFuture;
       });
+
+      Cache<K, V> cache = null;
       try {
          if (oldFuture != cacheFuture) {
-            return (Cache<K, V>) oldFuture.join();
+            cache = (Cache<K, V>) oldFuture.join();
+            if (!cache.getStatus().isTerminated()) {
```

Comment:
Here's the trick. If the cache exists but is stopped we do not return it, we proceed with create which will restart it.